src/dns/res.c

```diff
@@ -18,6 +18,12 @@
 #include <re_dns.h>
 #include "dns.h"
 
+#if defined(__linux__) && !defined(__GLIBC__) &&	\
+	!defined(__UCLIBC__) &&	!defined(__BIONIC__)
+/* Another popular libc for Linux offers OpenBSD's res_init() API */
+#define OPENBSD OPENBSD
+#endif
+
```

Comment:
this logic is not robust. what if there is a new libc project that is called e.g. MUSL_NG ?

you can work around this by building with the `HAVE_RESOLV` flag unset.